sdk/src/azure/core/az_mqtt5_rpc_client.c

```diff
@@ -0,0 +1,143 @@
+  _az_PRECONDITION_VALID_SPAN(client->_internal.client_id, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client->_internal.command_name, 1, false);
+  *out_topic_length = az_span_size(client->_internal.model_id)
+      + az_span_size(client->_internal.client_id) + az_span_size(client->_internal.command_name)
+      + 29;
+  _az_PRECONDITION_VALID_SPAN(out_subscription_topic, *out_topic_length, true);
+  // #endif
+
+  out_subscription_topic = az_span_copy(out_subscription_topic, AZ_SPAN_FROM_STR("vehicles/"));
+  out_subscription_topic = az_span_copy(out_subscription_topic, client->_internal.model_id);
+  out_subscription_topic = az_span_copy(out_subscription_topic, AZ_SPAN_FROM_STR("/commands/+/"));
+  out_subscription_topic = az_span_copy(out_subscription_topic, client->_internal.command_name);
+  out_subscription_topic = az_span_copy(out_subscription_topic, AZ_SPAN_FROM_STR("/__for_"));
+  out_subscription_topic = az_span_copy(out_subscription_topic, client->_internal.client_id);
+  out_subscription_topic = az_span_copy_u8(out_subscription_topic, '\0');
+
+  return AZ_OK;
+}
+
+// "vehicles/dtmi:rpc:samples:vehicle;1/commands/vehicle03/unlock/__for_mobile-app"
+AZ_NODISCARD az_result az_rpc_client_get_response_topic(
+    az_mqtt5_rpc_client* client,
+    az_span server_client_id,
+    az_span out_response_topic)
+{
+#ifndef AZ_NO_PRECONDITION_CHECKING
+  _az_PRECONDITION_VALID_SPAN(client->_internal.subscription_topic, 1, false);
+  _az_PRECONDITION_VALID_SPAN(server_client_id, 1, false);
+  int32_t response_topic_min_length
+      = az_span_size(client->_internal.subscription_topic) + az_span_size(server_client_id) - 1;
+  _az_PRECONDITION_VALID_SPAN(out_response_topic, response_topic_min_length, true);
+#endif
+
+  az_span_fill(out_response_topic, ' ');
+
+  int32_t index = az_span_find(client->_internal.subscription_topic, AZ_SPAN_FROM_STR("+"));
+  if (index > 0)
+  {
+    out_response_topic = az_span_copy(
+        out_response_topic, az_span_slice(client->_internal.subscription_topic, 0, index));
+    out_response_topic = az_span_copy(out_response_topic, server_client_id);
+    out_response_topic = az_span_copy(
+        out_response_topic, az_span_slice_to_end(client->_internal.subscription_topic, index + 1));
+  }
+  else
+  {
+    return AZ_ERROR_ITEM_NOT_FOUND;
+  }
+
+  return AZ_OK;
+}
+
+// "vehicles/dtmi:rpc:samples:vehicle;1/commands/vehicle03/unlock"
+AZ_NODISCARD az_result az_rpc_client_get_request_topic(
+    az_mqtt5_rpc_client* client,
+    az_span server_client_id,
+    az_span out_request_topic)
+{
+#ifndef AZ_NO_PRECONDITION_CHECKING
+  _az_PRECONDITION_VALID_SPAN(client->_internal.model_id, 1, false);
+  _az_PRECONDITION_VALID_SPAN(client->_internal.command_name, 1, false);
+  _az_PRECONDITION_VALID_SPAN(server_client_id, 1, false);
+  int32_t request_topic_min_length = az_span_size(client->_internal.model_id)
+      + az_span_size(server_client_id) + az_span_size(client->_internal.command_name) + 23;
```

Comment:
what is 23?

Reply:
This will be un-magic numbered in my PR to handle custom topic formats. It's the length of the topic format that's static (vehicles//commands//"